### DISCUSSION TITLE
System.TimeoutException: 'Could not retrieve latest GPS coordinates to be able to determine geofence current state' for iOS

### DISCUSSION THREAD

**Discussion by hannerygge** [shinyorg/shiny#1543] [Q&A]
When I try to use geofencing on an iPhone it fails. The OnStatusChanged is never triggered, and when I try to check the state of a geofence that is monitored I get the exception: System.TimeoutException: 'Could not retrieve latest GPS coordinates to be able to determine geofence current state'.

I have enabled all the permissions required and also the backgroundmode for location, also added the UseShiny() and the builder.Services.AddGeofencing<GeofenceDelegates>() have been added to MauiProgram.
This does also happen when I create a new project using the Shiny Template for Visual Studio.

A note though, this only happens on iOS, on Android it works as expected.

**Comment by aritchie** [comment]
You have to use the proper channel to file issues.  I know geofences work because I run them on my phone right now, so I need to see what you might be missing.

As for the exception, that can happen when the GPS can't get an immediate fix and you're requesting the current state of a geofence.
